### PR TITLE
[WIP] [Node] Add rough spec for Node JSON-RPC API

### PR DESCRIPTION
**Pull request by mvanderh**
### Description

Solidify 
- the methods and events that a Node is expected to implement
- error cases

Such that CF.js and the Node can be implemented in parallel while being sure that their interfaces will be compatible.

### CODE REVIEW

File: packages/node-provider/PROTOCOL_SPEC.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Protocol Spec
+
+## Public JSON-RPC Methods
+
+| method         | params                                                       | result                                                                  |
+|----------------|--------------------------------------------------------------|-------------------------------------------------------------------------|
+| cf_getApps     | []                                                           | {id, definition, state, asset, myDeposit, peerDeposit, timeout}[]       |
+| cf_install     | [definition, state, asset, myDeposit, peerDeposit, timeout]  | {id}                                                                    |
```
Comment (cf19drofxots) [inline packages/node-provider/PROTOCOL_SPEC.md]: What about proposeInstall?

Comment (cf19drofxots) [reply packages/node-provider/PROTOCOL_SPEC.md]: Based on https://github.com/counterfactual/monorepo/blob/master/packages/cf.js/API_REFERENCE.md

Seems like `proposeInstall` would return the ID of the AppInstance, which would _then_ be used to make the actual install upon acceptance, correct?

**Comment by joelalejandro** [comment]
@mvanderh Should there be any methods related to propose/accept/reject state? I'm not sure if `stateUpdate` represents those messages.

File: packages/node-provider/PROTOCOL_SPEC.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+TODO
+
+
+
+## Events
+
+| event          | params                                                          |
+|----------------|-----------------------------------------------------------------|
+| install        | {id, definition, state, asset, myDeposit, peerDeposit, timeout} |
```
Comment (snario) [inline packages/node-provider/PROTOCOL_SPEC.md]: Need type information

File: packages/node-provider/PROTOCOL_SPEC.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- `id`: set to a generated unique ID for this request
+- `method`: name of the method to call
+- `params`: array of parameters passed to the method call  
+
+## Methods
+
+| method         | params                                                       | result                                                                  |
+|----------------|--------------------------------------------------------------|-------------------------------------------------------------------------|
+| cf_getApps     | []                                                           | list of {id, definition, state, asset, myDeposit, peerDeposit, timeout} |
+| cf_install     | [definition, state, asset, myDeposit, peerDeposit, timeout]  | {id}                                                                    |
```
Comment (snario) [inline packages/node-provider/PROTOCOL_SPEC.md]: need type information

**Pull request closed without merging**